docs/source/transaction_family_specifications/identity_transaction_family.rst

```diff
@@ -0,0 +1,217 @@
+
+Address construction for roles will follow a pattern similar to address
+construction in the settings namespace. Role names will be broken into four
+parts, where parts of the string are delimited by the "." character. For
+example, the key a.b.c would be split into the parts "a", "b", "c", and the
+empty string. If a key would have more than four parts the extra parts are left
+in the last part. For example, the key a.b.c.d.e would be split into "a", "b",
+"c", and "d.e".
+
+A short hash is computed for each part. For the first part the first 14
+characters of the SHA-256 hash are used. For the remaining parts the first 16
+characters of the SHA-256 hash are used. The address is formed by concatenating
+the identity namespace “00001d”, the role namespace “01”, and the four short
+hashes.
+
+For example, the address for the role client.query_state would be constructed
+as follows:
+
+.. code-block:: pycon
+
+  >>> "00001d"+ "01" + hashlib.sha256('client'.encode()).hexdigest()[:14]+ \
+    hashlib.sha256('query_state'encode()).hexdigest()[:16]+ \
+    hashlib.sha256(''encode()).hexdigest()[:16]+ \
+    hashlib.sha256(''encode()).hexdigest()[:16]
+
+Transaction Payload
+===================
+Identity transaction family payloads are defined by the following protocol
+buffers code:
+
+File: sawtooth-core/families/identity/protos/identity.proto
+
+.. code-block:: protobuf
+
+  message IdentityPayload {
+      enum IdentityType {
+        POLICY = 0;
+        ROLE = 1;
+      }
+
+      // Which type of payload this is for
+      IdentityType type = 1;
+
+      // Serialize bytes of a role or a policy
+      bytes data = 2;
+  }
+
+Transaction Header
+==================
+
+Inputs and Outputs
+------------------
+
+The inputs for config family transactions must include:
+
+* the address of *sawtooth.identity.allowed_keys*
+* the address of the role or policy being changed
+* if setting a role, the address of the policy to assign to the role
+
+The outputs for config family transactions must include:
+
+* the address of the role or policy being changed
+
+Dependencies
+------------
+
+None.
+
+Family
+------
+
+- family_name: "sawtooth_identity"
+- family_version: "1.0"
+
+Encoding
+--------
+
+The encoding field must be set to "application/protobuf".
+
+Execution
+=========
+Initially, the transaction processor gets the current values of
+sawtooth.identity.allowed_keys from the state.
+
+The public key of the transaction signer is checked against the values in the
+list of allowed keys. If it is empty, no roles or policy can be updated. If
+the transaction signer is not in the allowed keys the transaction is invalid.
+
+Whether this is a role or a policy transaction is checked by looking at the
+IdentityType in the payload.
+
+If the transaction is for setting a policy, the data in the payload will be
+parsed to form a policy object. The policy object is then checked to make sure
```

Comment:
Suggest: 
```
form a ``Policy`` object.
```

Other uses referring specifically to an in-code class should follow the same formatting.